apps/web/pages/api/verify-booking-token.ts

```diff
@@ -0,0 +1,91 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { authenticator } from "otplib";
+import { z } from "zod";
+
+import { defaultResponder } from "@calcom/lib/server";
+import prisma from "@calcom/prisma";
+import { UserPermissionRole } from "@calcom/prisma/enums";
+import { TRPCError } from "@calcom/trpc/server";
+import { createContext } from "@calcom/trpc/server/createContext";
+import { bookingsRouter } from "@calcom/trpc/server/routers/viewer/bookings/_router";
+import type { UserProfile } from "@calcom/types/UserProfile";
+
+enum DirectAction {
+  ACCEPT = "accept",
+  REJECT = "reject",
+}
+
+const querySchema = z.object({
+  action: z.nativeEnum(DirectAction),
+  token: z.string(),
+  reason: z.string().optional(),
+  bookingUid: z.string(),
+  userId: z.string(),
+});
+
+async function handler(req: NextApiRequest, res: NextApiResponse<Response>) {
+  const { action, token, reason, bookingUid, userId } = querySchema.parse(req.query);
```

Comment:
How are we getting the reason here? We cannot provide it from email AFAIK. Unless que put an [actual form in the email](https://www.mailmodo.com/guides/html-form-emails/) and transform this endpoint into a POST request.

Reply:
Yeah, it's not possible now. I missed it to remove